File: spring-webflux/src/main/java/software/darkmatter/school/blog/domain/user/business/UserService.java
package software.darkmatter.school.blog.domain.user.business;

import org.springframework.data.domain.Pageable;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;
import software.darkmatter.school.blog.api.dto.UserCreateDto;
import software.darkmatter.school.blog.domain.user.data.User;

import java.util.List;

public interface UserService {

    Flux<User> getList(Pageable pageable);

    Flux<User> getListByIds(List<Long> ids);

    Mono<User> getById(Long id);

    Mono<User> create(UserCreateDto userCreateDto);

    Mono<User> update(Long id, UserCreateDto userCreateDto);

    Mono<Void> delete(Long id);
}

File: kotlin-spring-webflux/src/main/kotlin/software/darkmatter/school/blog/domain/user/business/UserService.kt
package software.darkmatter.school.blog.domain.user.business

import org.springframework.data.domain.Pageable
import software.darkmatter.school.blog.api.dto.UserCreateDto
import software.darkmatter.school.blog.domain.user.domain.User

interface UserService {
    suspend fun getList(pageable: Pageable): List<User>
    suspend fun getListByIds(ids: List<Long>): List<User>
    suspend fun getById(id: Long): User
    suspend fun create(userCreateDto: UserCreateDto): User
    suspend fun update(id: Long, userCreateDto: UserCreateDto): User
    suspend fun delete(id: Long)
}
